jobs/jobscheduler.py

```diff
@@ -358,6 +375,11 @@ def initJobs(jobs, resources, otype):
     }
     print(MachineCPUCount, MachineMemoryGB, resources)
     jobs = initJobs(json.load(open(opts.jobs)), resources, opts.type)
+
+    needGPU = any("_VISIBLE_DEVICES" in c["command"] for j in jobs["jobs"] for c in j["commands"])
```

Comment:
why not just
```
needGPU = any("gpu" in c for j in jobs["jobs"] for c in j["commands"])
```

Reply:
Because there is no `gpu`  in command, e.g. `cd 141.044412_Run3-2023_JetMET2023D_RecoECALOnlyGPU; CUDA_VISIBLE_DEVICES=0 HIP_VISIBLE_DEVICES= cmsDriver.py step2  --process reHLT -s L1REPACK:Full,HLT:@relval2023 --conditions auto:run3_hlt_relval --data  --eventcontent FEVTDEBUGHLT --datatier FEVTDEBUGHLT --era Run3_2023  --customise Validation/Performance/TimeMemorySummary.customiseWithTimeMemorySummary --prefix 'python2 .../monitor_workflow.py timeout --signal SIGTERM 9000 '  -n 100  --filein filelist:step1_dasquery.log --lumiToProcess step1_lumiRanges.log --fileout file:step2.root  --suffix \"-j JobReport2.xml \"  --nThreads 4 > step2_Run3-2023_JetMET2023D_RecoECALOnlyGPU.log  2>&1`

Reply:
`j["command"]` is a dict and we add `gpu` in it in the `initJobs` function which was called just above this line